simple/utils/spectra_convert.py

```diff
@@ -139,50 +120,75 @@ def compile_header(wavelength_data, **original_header_dict):
+
     try:
         bandpass = original_header_dict["bandpass"]
         header.set("SPECBAND", bandpass, "SED.bandpass")
     except KeyError:
-        bandpass = None
-        logging.warning("No bandpass")
+        bandpass = assign_ucd(w_mid * wavelength_data.unit)
+        header.set("SPECBAND", bandpass, "SED.bandpass")
 
+    # OPTIONAL Header keywords
     try:
-        header.set("SPEC_VAL", w_mid, f"[{w_units}] Characteristic spec coord")
+        exposure_time = original_header_dict["exposure_time"]
+        header.set("TELAPSE", exposure_time, "[s] Total elapsed time")
     except KeyError:
-        pass
+        exposure_time = input("Please input the exposure time in seconds: ")
+        if exposure_time != "":
+            header.set("TELAPSE", exposure_time, "[s] Total elapsed time")
 
-    header.set("SPEC_BW", width, f"[{w_units}] Width of spectrum")
-    header.set("TDMIN1", w_min, f"[{w_units}] Starting wavelength")
-    header.set("TDMAX1", w_max, f"[{w_units}] Ending wavelength")
+    try:
+        time_start = Time(dateparser.parse(original_header_dict["start_time"])).jd
+        header.set("TSTART", time_start, "[d] MJD start time")
+    except KeyError:
+        time_start = None
 
     try:
-        aperture = original_header_dict["aperture"]
-        header.set("APERTURE", aperture, "[arcsec] slit width")
+        time_stop = Time(dateparser.parse(original_header_dict["stop_time"])).jd
+        header.set("TSTOP", time_stop, "[d] MJD stop time")
     except KeyError:
-        aperture = None
-        logging.warning("No aperature")
+        time_stop = None
 
     try:
-        obs_date = to_datetime(original_header_dict["obs_date"]).strftime("%Y-%m-%d")
-        header.set("DATE-OBS", obs_date, "date of the observation")
+        time = (
+            Time(dateparser.parse(original_header_dict["start_time"])).jd
+            + Time(dateparser.parse(original_header_dict["stop_time"])).jd
+        ) / 2
+        header.set("TMID", time, "[d] MJD mid expsoure")
+    except KeyError:
+        time = None
+
+    try:
+        obs_location = original_header_dict["observatory"]
+        header.set("OBSERVAT", obs_location, "name of observatory")
     except KeyError:
-        print(KeyError)
-        obs_date = None
-        logging.warning("No observation date")
+        obs_location = None
+
+    try:
+        aperture = original_header_dict["aperture"]
+        header.set("APERTURE", aperture, "[arcsec] slit width")
+    except KeyError:
+        aperture = input("OPTIONAL: Please input the slitwidth in arcseconds: ")
+        if aperture != "":
+            header.set("APERTURE", aperture, "[arcsec] slit width")
 
     # Publication Information
     try:
         title = original_header_dict["title"]  # trim so header wraps nicely
         header.set("TITLE", title, "Data set title")
     except KeyError:
-        pass
-        logging.warning("No title")
+        title = None
```

Comment:
As an alternative to so many try..except blocks, consider using something like:
```python
title = original_header_dict.get("title")
if title is not None:
   header.set("TITLE", title, "Data set title")
```
The python dictionary's `get` method automatically provides a default value of `None` if the key is not present: https://docs.python.org/3.10/library/stdtypes.html#dict.get